include/pybind11/trampoline_self_life_support.h

```diff
@@ -4,9 +4,13 @@
 
 #pragma once
 
-#include "detail/common.h"
-#include "detail/smart_holder_poc.h"
-#include "detail/value_and_holder.h"
+#include "detail/internals.h"
+
+#ifdef PYBIND11_HAVE_INTERNALS_WITH_SMART_HOLDER_SUPPORT
+
+#    include "detail/common.h"
```

Comment:
Do these need to be ifdef guarded?

Reply:
It's a choice (for the most part at least).

I did it that way so that it is as obvious as possible that pybind11 behaves almost exactly as before when `PYBIND11_HAVE_INTERNALS_WITH_SMART_HOLDER_SUPPORT` is not defined.

(The few tiny exceptions are a compromise, `#ifdef` clutter vs. tiny pieces from the smart_holder code leaking in.)

I prefer to keep this as-is, unless you believe it's better to not have conditional `#include`s.